cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -211,25 +217,35 @@ Status ExtractDecimalMinMaxFromBytesType(const Statistics& statistics,
                                          std::shared_ptr<::arrow::Scalar>* max) {
   const DecimalLogicalType& decimal_type =
       checked_cast<const DecimalLogicalType&>(logical_type);
-
-  Result<std::shared_ptr<DataType>> maybe_type =
-      Decimal128Type::Make(decimal_type.precision(), decimal_type.scale());
+  auto precision = decimal_type.precision();
+  auto scale = decimal_type.scale();
   std::shared_ptr<DataType> arrow_type;
-  if (maybe_type.ok()) {
-    arrow_type = maybe_type.ValueOrDie();
+
+  if (precision <= Decimal32Type::kMaxPrecision) {
```

Comment:
Can we add a comment that the literal would can be cast to the correspond type if the real reader type is a wider decimal type?